pay/example/lib/main.dart

```diff
@@ -152,6 +168,25 @@ class _PaySampleAppState extends State<PaySampleApp> {
               child: CircularProgressIndicator(),
             ),
           ),
+          // This allows you to control whether the payment shall succeed or not.
+          // This is only available on iOS.
+          // Usually you would want to wait for your backend to return a value,
+          // before you update the payment status.
+          if (defaultTargetPlatform == TargetPlatform.iOS)
```

Comment:
The goal of this sample app is to show a regular integration path using the plug-in. The intention is not to build an application that tests the potential available scenarios. My suggestion is to remove this block, and add a comment in the `onApplePayResult` callback below the `debugPrint` that reminds users of the library that this is where they'd want to validate the response, communicate with their servers to confirm the transaction and return errors if any.

Below that, I'd simulate a successful result: `await _pay.updatePaymentStatus(true);`